cf/docstring/docstring.py

```diff
@@ -613,6 +621,37 @@
     "{{to_size: `int`, optional}}": """to_size: `int`, optional
                 Pad the axis after so that the new axis has the given
                 size.""",
+    # subspace mode options
+    "{{mode: optional}}": """mode: optional
```

Comment:
I think it's because if you provide just one positional parameter, it could be the mode _or_  the halo. Perhaps a name different to "mode" would be better?

Reply:
like `config`?

Reply:
Then the `config` could define the "mode" and/or the "halo"

Reply:
Ah, I see. That explanation makes sense, though (might be missing something here), why do we need to stick to positional arguments in the first place, as opposed to demoting one (in a sense) to a keyword argument? 

That said, I like your idea of keeping them but changing the name to config, in line with the new combined purpose of the keyword. As long as you don't mind changing the API that way for this PR and release, I'd favour that.

Reply:
> why do we need to stick to positional arguments in the first place, as opposed to demoting one (in a sense) to a keyword argument?

This is because every conceivable keyword argument has to be reserved for axis identification.

Reply:
Done: cb535c45235dd525d9414d270875974ab7f1a361